src-ghc/Pact/Interpreter.hs

```diff
@@ -194,13 +198,54 @@ setupEvalEnv dbEnv ent mode msgData refStore gasEnv np spv pd ec = do
   , _eeWarnings = warnRef
   }
   where
+    versionNatives = appEndo $ mconcat
+      [ disablePact40Natives ec
+      , disablePact420Natives ec
+      , disablePact43Natives ec
+      , disablePact431Natives ec
+      , disablePact46Natives ec ]
     mkMsgSigs ss = M.fromList $ map toPair ss
       where
         toPair Signer{..} = (pk,S.fromList _siCapList)
           where
             pk = PublicKeyText $ fromMaybe _siPubKey _siAddress
 
 
+setupNativesEvalEnv
```

Comment:
So this is an interesting one.

I think this function is what chainweb and co. should use, as it manages the natives forking.

Probably for all "Default" deployments of pact, there's no need for a downstream user to be using `initRefStore`. I looked across all callsites in chainweb and pact and that parameter was always the same.

I think we should leave `setupEvalEnv` as is, as most of the other parameters do usually get tweaked (E.g free gas models, db, etc), but maybe we should also perform more of this forking in pact (e.g gas model forking) in this function, and simply have chainweb call a `setupEvalEnvChainweb` that we expose specifically for chainweb. That way, pactservice only ever has to worry about when to introduce the flags, and not about semantic changes in the pact interpreter.

Thoughts? @sirlensalot @jwiegley @emilypi

Reply:
Why not just provide a `versionedNativesRefStore` for chainweb to switch to, and leave `setupEvalEnv` alone?

Reply:
I tend to agree with @sirlensalot here. Can't Chainweb still choose what eval env it's setting up if we allow it to parametrize `setupEvalEnv` with a versioned native refstore? We already build our own env here: https://github.com/kadena-io/chainweb-node/blob/2dcdc28b6ccfa2102a16051165ed5d4e100b9084/src/Chainweb/Pact/TransactionExec.hs#L1123. 

Seems like we just want to change `initRefStore` to `chainwebVersionedRefStore` or smth.

Reply:
The current setup is more flexible and "less surprising". Your approach bakes in a lot of behavior into a function that is otherwise a smart constructor for `EvalEnv`, which may create unpleasant surprises for environments that don't need to suppress new natives.

If chainweb always calls it one way it can solve any duplication trivially with factoring in the chainweb codebase.

The desired behavior affects `RefStore`, not `EvalEnv`, so the burden of argument is to justify why operations on the outer datastructure need to change for a single member. The most straightforward approach is to offer a versioned `RefStore` and let clients choose to use it.

Reply:
I made a `versionedNativesRefStore`.